src/app/download/download.component.ts

```diff
@@ -1,32 +1,37 @@
 import {Component} from '@angular/core';
-import {AdAuthoringService} from '../ad-authoring/ad-authoring.service';
-import {Observable} from 'rxjs';
-import {map} from 'rxjs/operators';
 import {DownloadService} from './download.service';
+import * as JSZip from 'jszip';
 
 @Component({
   selector: 'app-download',
   templateUrl: './download.component.html',
   styleUrls: ['./download.component.scss'],
 })
 export class DownloadComponent {
-  fileUrl;
-  ampHtmlObs: Observable<string>;
-
   constructor(private service: DownloadService) {}
 
-  downloadFile() {
+  downloadFileZip() {
+    const zip = new JSZip();
     const data = this.service.generateHtmlForDownload();
-    // const data = 'amp story ad html goes here';
-    const a = document.createElement('a');
-    const blob = new Blob([data], {type: 'application/octet-stream'});
-    const url = window.URL.createObjectURL(blob);
-    document.body.appendChild(a);
-    a.style.display = 'none';
-    a.href = url;
-    a.download = 'file.txt';
-    a.click();
-    window.URL.revokeObjectURL(url);
-    document.body.removeChild(a);
+    zip.file('AmpAd.html', data);
+
+    const asset = this.service.getAsset();
+    const assetBase64 = asset[0];
+    // strip off everything befor the first comma (data:image/png;base64,) to get raw base64 encoding
+    const assetBase64Stripped = assetBase64.replace(/^[^,]+, */, '');
+    const assetFileName = asset[1];
+    zip.file(assetFileName, assetBase64Stripped, {base64: true});
+
+    zip.generateAsync({type: 'blob'}).then(function (content) {
+      const objectUrl: string = URL.createObjectURL(content);
+      const a: any = document.createElement('a');
```

Comment:
Remove the 'any' type since it doesn't give us any valuable type information here. Also rename to something more descriptive i.e.

```suggestion
      const downloadLink = document.createElement('a');
```